### ISSUE TITLE
Investigate test failures

### ISSUE THREAD

**Issue by anghelcovici** [project-oak/oak#200]
I've seen test failing with:
QFATAL  manager_client.h : 51 : Failed: 14/Socket closed/

This seems to be race condition, it needs some investigation.

**Comment by anghelcovici** [comment]
cc: @tiziano88

**Comment by daviddrysdale** [comment]
The intermittent failures where `//oak/server/asylo:oak` crashes seem to be getting more common.

Notes so far:
  - It doesn't seem to happen with the dev server (I tweaked a Travis build to use `//oak/server/dev:oak` and it succeeded 20 times in a row).
  - I have failed to reproduce the crash in a local Docker run, and we haven't observed it in the Google Cloud Build, so it seems to be specific to Travis+Asylo.
  - I've run the (large) `abitest` example locally against the dev server under TSAN, ASAN and UBSAN without anything turning up.
  - I experimented with trying to get a backtrace and symbolize it, but under Asylo the symbols aren't attached and under Travis+Asylo the backtrace only had [a single entry](https://travis-ci.org/daviddrysdale/oak/jobs/595722407) (`2019-10-09 17:18:55  ERROR  asylo_oak_main.cc : 52 : backtrace[0] = 0x7f12155cd730 __restore_rt`)
